modules/qna/src/views/full/Components/TextareaList.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button } from '@blueprintjs/core'
+// @ts-ignore
+import BotpressContentPicker from 'botpress/content-picker'
+// @ts-ignore
+import BotpressContentTypePicker from 'botpress/content-type-picker'
+import { lang, Textarea } from 'botpress/shared'
+import _uniqueId from 'lodash/uniqueId'
+import React, { FC, Fragment, useRef, useState } from 'react'
+
+import style from '../style.scss'
+
+interface Props {
+  updateItems: (items: any) => void
+  items: string[]
+  placeholder: (index: number) => void
+  addItemLabel: string
+  label: string
+  keyPrefix: string
+  showPicker?: boolean
+  initialFocus?: string
+  canAddContent?: boolean
+}
+
+const TextareaList: FC<Props> = props => {
+  const [showPicker, setShowPicker] = useState(false)
+  const focusedElement = useRef(props.initialFocus || '')
+  const { updateItems, keyPrefix, canAddContent, addItemLabel, label, items, placeholder } = props
+
+  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
+  const keys = useRef([...Array(items.length)].map(x => _uniqueId(keyPrefix)))
+
+  const updateItem = (index, value) => {
+    items[index] = value
+    updateItems(items)
+  }
+
+  const addItem = (value = '') => {
+    items.push(value)
+    keys.current.push(_uniqueId(keyPrefix))
+    focusedElement.current = `${keyPrefix}${items.length - 1}`
+    updateItems(items)
+  }
+
+  const onKeyDown = (e, index) => {
+    if (e.key === 'Enter' && e.shiftKey) {
+      e.preventDefault()
+      addItem()
+    }
+
+    if (e.key === 'Backspace' && items.length > 1 && !items[index].length) {
+      e.preventDefault()
+
+      items.splice(index, 1)
```

Comment:
I'm not sure I understand, backspace deletes a whole item ?

Reply:
only when there's no content in it